File: payloads/c/clone_newuser.c
#define _GNU_SOURCE
#include <stdio.h>
#include <sched.h>

int main(void) {
    if (unshare(CLONE_NEWUSER) != 0) {
        perror("unshare CLONE_NEWUSER blocked");
        return 1;
    }
    printf("ESCAPED\n");
    return 0;
}

File: payloads/go/clone_newuser.go
package main

import (
	"fmt"
	"syscall"
)

func main() {
	_, _, errno := syscall.RawSyscall(syscall.SYS_UNSHARE, syscall.CLONE_NEWUSER, 0, 0)
	if errno != 0 {
		panic(fmt.Sprintf("unshare failed: %v", errno))
	}
	fmt.Println("ESCAPED")
}
